### ISSUE TITLE
Add utilities for measuring an effect's execution time

### ISSUE THREAD

**Issue by kubukoz** [typelevel/cats-effect#1672]
Something of the sort:

```scala
//in IO[A]

def timed: IO[(FiniteDuration, A)]

//in GenTemporal syntax

def timed: F[(FiniteDuration, A)]
```

After it lands in series/3.x, I think it would be a welcome addition to CE2 as well (for effects that have a `Clock[F]`, maybe - it would require a new syntax class).

**Comment by patseev** [comment]
Will take this one 🙂

**Comment by kubukoz** [comment]
Closed in #1677
